events/apigw.go

```diff
@@ -291,15 +291,8 @@ type APIGatewayV2CustomAuthorizerSimpleResponse struct {
 	Context      map[string]interface{} `json:"context,omitempty"`
 }
 
-// APIGatewayCustomAuthorizerPolicy represents an IAM policy
-type APIGatewayCustomAuthorizerPolicy struct {
-	Version   string
-	Statement []IAMPolicyStatement
-}
-
-// IAMPolicyStatement represents one statement from IAM policy with action, effect and resource
-type IAMPolicyStatement struct {
-	Action   []string
-	Effect   string
-	Resource []string
-}
+// APIGatewayCustomAuthorizerPolicy represents an IAM policy.
+//
+// Deprecated: This type exists for backwards compatibility. New events
+// and structs should reference IAMPolicyDocument directly instead.
+type APIGatewayCustomAuthorizerPolicy IAMPolicyDocument
```

Comment:
To keep backcompat, I believe this would need to be [a type alias](https://go.googlesource.com/proposal/+/master/design/18130-type-alias.md). eg: write as `type APIGatewayCustomAuthorizerPolicy = IAMPolicyDocument`

Reply:
That is not necessary for backwards compatibility. `type A B` makes A a copy of B (same struct fields), they just are not equivalent/interchangeable. `type A = B` makes A an alias of B, meaning they are interchangeable (e.g., `type byte = uint8`). Unless you want clients to be able to use an `APIGatewayCustomAuthorizerPolicy` wherever an `IAMPolicyDocument` is expected without casting, there is no need for an alias.

Reply:
my brain read the diff as having `APIGatewayCustomAuthorizerResponse` also being updated to use IAMPolicyDocument for it's `PolicyDocument` field, which would have broken something like `x.PolicyDocument = APIGatewayCustomAuthorizerPolicy{}` (see: https://play.golang.org/p/_OXhN9ebJ1K). I see now that `APIGatewayCustomAuthorizerResponse` wasn't updated. I can't tell if that was intentional or not?

If you think that `APIGatewayCustomAuthorizerResponse` shouldn't be updated, then lets remove the `//Deprecated:` comment, as it's use of `APIGatewayCustomAuthorizerPolicy` would still be correct in that context

Reply:
The "deprecated" comment was directed more at API developers (to tell them not to use it in new structs), not API clients.
I had not intended to make changes to `APIGatewayCustomAuthorizerResponse` but I can do that if you prefer.

Reply:
I think to just drop the comment so that no one's linter freaks out. DRYing the rest api gateway type feels out of scope to me for this PR